From 25aeeb6250dc5f6ad508380ae447f2de936fa9c7 Mon Sep 17 00:00:00 2001
From: Marek Marczykowski <marmarek@invisiblethingslab.com>
Date: Tue, 23 Apr 2013 05:25:00 +0200
Subject: [PATCH 106/106] libvchan: do not use xc_gntshr_share_page_notify in
 Mini-OS
Organization: Invisible Things Lab
Cc: Marek Marczykowski <marmarek@invisiblethingslab.com>

It isn't implemented there yet.

Signed-off-by: Marek Marczykowski <marmarek@invisiblethingslab.com>
---
 tools/libvchan/init.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/tools/libvchan/init.c b/tools/libvchan/init.c
index 8a66cef..54aa08e 100644
--- a/tools/libvchan/init.c
+++ b/tools/libvchan/init.c
@@ -78,9 +78,14 @@ static int init_gnt_srv(struct libxenvchan *ctrl, int domain)
 	uint32_t ring_ref = -1;
 	void *ring;
 
+#ifdef __MINIOS__
+	/* _notify part not implemented in Mini-OS */
+	ring = xc_gntshr_share_pages(ctrl->gntshr, domain, 1, &ring_ref, 1);
+#else
 	ring = xc_gntshr_share_page_notify(ctrl->gntshr, domain,
 			&ring_ref, 1, offsetof(struct vchan_interface, srv_live),
 			ctrl->event_port);
+#endif
 
 	if (!ring)
 		goto out;
-- 
1.8.1.4

